pulsar/client_impl_test.go

```diff
@@ -67,6 +67,24 @@ func TestTLSInsecureConnection(t *testing.T) {
 	client.Close()
 }
 
+func TestTLSInsecureConnectionWithCerts(t *testing.T) {
```

Comment:
This test doesn't seem to do anything substantive.  I'd expect it to exercise the resultant `tls.Config` in some way, maybe by making an actual TLS connection.